sisyphus/ci.go

```diff
@@ -161,24 +162,38 @@ func (p *ProwAccessor) GetResult(jobName string, runNo int) (*Result, error) {
 		log.Printf("Failed to unmarshal ProwResult %s: %v", prowResultString, err)
 		return nil, err
 	}
-	return &Result{
-		Passed: prowResult.Passed,
-		SHA:    prowResult.Metadata.RepoCommit,
-	}, nil
+	cfg, err := p.getProwJobConfig(jobName, runNo)
+	if err != nil {
+		return nil, err
+	}
+	// Presubmit
+	// "master:241d69701ebf3c57c0d86d016937d23615dae390,554:cc02e59987163c8deb80db8bcc203c318c1b752e"
+	// Postsubmit
+	// "master:241d69701ebf3c57c0d86d016937d23615dae390"
+	for _, baseSHAprSHA := range cfg.Repos {
```

Comment:
Can you just check the length of cfg.Repos and take the first element to make future review easier ?

Reply:
I totally agree with you but it seems impossible to access a key-value pair in go map using an index. For example https://groups.google.com/forum/#!topic/golang-nuts/zBcqMsDNt7Q. The return statement in the for loop ensures we do one iteration, and the return statement after the loop generates unfound exception.

Reply:
I did not realized it was a map. That brings another issue I guess, how do you know the SHA is the right one ? Is it because in our case the map length is always one ? if that the case please make it as an assumption and fail otherwise.

Reply:
Yes, the map length is always one. I added an explicit check for this and fail otherwise as you suggested. Also added more comments explaining the structure of started.json so people looking at my code have better context of what it is doing. PTAL